### ISSUE TITLE
2025 W1 Notes

### ISSUE THREAD

**Issue by gpleiss** [UBC-STAT/stat-406#23]
Running list of notes/proposed changes for next year:

- Cut down on course overview slides, so there's more time for probability review

## Module 0

- Include review of SVD!!!
- After students derive the tower property, consider having the students derive the law of total variance as a rule that students derive? Or get students used to conditional variance notation.

## Module 1

- Tower rule/conditional expectations take a bit of time
- Maybe ignore algorithmic perspective
- When introducing the statistical perspective, start off with model/prediction/OLS estimator to establish what we know, then talk about deriving them
- Key element: what we need is (1) a statistical model (2) a loss function and (3) an estimation procedure
- Weaving in classification/logistic regression with linear regression seems to work decently
- Remove "unbiased" language around cross-validation; they're not ready for it yet

### High level reorganization

- Make a separate classification module
- Save decision theory on prediction for this classification module.
- Disconnect prediction models from statistical models.

## Module 2

- Lasso: needs more intuitions behind it (like ridge), e.g. on how it favours sparsity. More plots would help.

## Module 3

- Knn: pictorially represent how bias decreases with the amount of data (e.g. piecewise constant regions). This was a point of confusion for students.
- Kernel ridge regression: include more examples of kernels, matching the basis functions that they learn about (e.g. polynomial). Do not attempt to derive how the RBF kernel corresponds to infinitely many Fourier basis features; just state without proof

## Module 4

Get rid of this module all together? Replace it with a module on classification and...

- Decision theory (e.g. different prediction rules - probabilistic, weighted, etc. - based on different notions of loss)
- Generative models (e.g. discuss LDA versus logistic regression)
- A classification model/logistic regression derivation would be the best place to talk about MLE
- We can cover gradient descent in this module before neural networks

## Grading

- Follow careful instructions with github setup (don't allow students to create repos, include rproj files in templates)
- Create student repos when they are invited to the github org, not after they join (this way they already have repos available to them when they join)
- Name the lab/hw repos after the student name or id number, rather than after their github id
- Have due dates for when grades should be in for the TAs
- Grade Lab 00/01 on the Monday after it's due to catch people who don't know how to use github
- Create a script that auto-downloads all submissions from github, and creates PR approval/comments based on info uploaded to a spreadsheet
- Ensure that TAs merge PRs after they create them
- Make sure grades on assignments are visible on Canvas
